.circleci/config.yml

```diff
@@ -168,8 +170,304 @@ jobs:
+          command: |
+            mv src/contract-artifacts.ts src/contract-artifacts._ts
+            mv src/contract-deployed-artifacts.ts src/contract-deployed-artifacts._ts
+          working_directory: packages/contracts
+      - run:
+          name: Lint
+          command: yarn lint:check
+          working_directory: packages/contracts
+      - run:
+          name: Rename TS files back
+          command: |
+            mv src/contract-artifacts._ts src/contract-artifacts.ts
+            mv src/contract-deployed-artifacts._ts src/contract-deployed-artifacts.ts
+          working_directory: packages/contracts
+      - run:
+          name: Slither
+          command: yarn test:slither
+          working_directory: packages/contracts
+      - run:
+          name: Test
+          command: yarn test:coverage
+          working_directory: packages/contracts
+
+  dtl-tests:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Test
+          command: yarn test:coverage
+          working_directory: packages/data-transport-layer
+
+  geth-tests:
+    docker:
+      - image: mslipper/go-builder:latest
+    steps:
+      - checkout
+      - run:
+          name: Test
+          command: make test
+          working_directory: l2geth
+
+  depcheck:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Check contracts
+          command: npx depcheck
+          working_directory: packages/contracts
+      - run:
+          name: Check core-utils
+          command: npx depcheck
+          working_directory: packages/core-utils
+      - run:
+          name: Check data-transport-layer
+          command: npx depcheck
+          working_directory: packages/data-transport-layer
+      - run:
+          name: Check sdk
+          command: npx depcheck
+          working_directory: packages/sdk
+      - run:
+          name: Check integration-tests
+          command: npx depcheck
+          working_directory: integration-tests
+
+  bss-core-tests:
+    docker:
+      - image: mslipper/go-builder:latest
+    steps:
+      - checkout
+      - run:
+          name: Lint
+          command: golangci-lint run -E goimports -E sqlclosecheck -E bodyclose -E asciicheck ./...
+          working_directory: go/bss-core
+      - run:
+          name: Test
+          command: |
+            mkdir -p /test-results
+            gotestsum --junitfile /test-results/tests.xml
+          working_directory: go/bss-core
+      - store_test_results:
+          path: /test-results
+
+  integration-tests:
+    machine:
+      image: ubuntu-2004:202111-02
+      docker_layer_caching: true
+    environment:
+      DOCKER_BUILDKIT: 1
+    parallelism: 3
```

Comment:
Where does the parallelism apply? In the `circleci tests` step below?

Reply:
Correct. `circleci tests` will split the test files based on filename, which will then be run across 3 executors.